ozone/core/models/data.py

```
@@ -518,6 +522,107 @@ class BaseImportExportReport(models.Model):
+                if blend:
+                    # Blend entry
+                    blend_subs = blend.get_substance_ids_percentages()
+                    for substance, percentage in blend_subs:
+                        if substance in related_substances:
+                            sums_dictionary[substance]['totals_sum'] += sum(
+                                [
+                                    getattr(entry, field, Decimal(0)) * percentage
+                                    for field in totals_fields
+                                ]
+                            )
+                            sums_dictionary[substance]['quantities_sum'] += sum(
+                                [
+                                    getattr(entry, field, Decimal(0)) * percentage
+                                    for field in quantity_fields
+                                ]
+                            )
+                elif substance:
+                    if substance.id in related_substances:
+                        sums_dictionary[substance.id]['totals_sum'] += sum(
+                            [
+                                getattr(entry, field, Decimal(0)) or Decimal(0)
+                                for field in totals_fields
+                            ]
+                        )
+                        sums_dictionary[substance.id]['quantities_sum'] += sum(
+                            [
+                                getattr(entry, field, Decimal(0)) or Decimal(0)
+                                for field in quantity_fields
+                            ]
+                        )
+
+            # And finally verify that, for each substance,
+            # sum of totals > sum of quantities
+            valid = all(
+                [
+                    sums['totals_sum'] >= sums['quantities_sum']
+                    for sums in sums_dictionary.values()
+                ]
+            )
+            if not valid:
```

You can write the `all` as a loop and throw an error at the first invalid value. That way you can provide more info to the user.

---

Indeed, done.